index.js

```diff
@@ -210,41 +188,20 @@ class TuyaDevice extends EventEmitter {
     debug('SET Payload:');
     debug(payload);
 
-    // Encrypt data
-    const data = this.device.cipher.encrypt({
-      data: JSON.stringify(payload)
-    });
-
-    // Create MD5 signature
-    const md5 = this.device.cipher.md5('data=' + data +
-      '||lpv=' + this.device.version +
-      '||' + this.device.key);
-
-    // Create byte buffer from hex data
-    const thisData = Buffer.from(this.device.version + md5 + data);
-
     // Encode into packet
-    const buffer = Parser.encode({
-      data: thisData,
-      commandByte: 7 // 0x07
+    const buffer = this.device.parser.encode({
+      data: payload,
+      encrypted: true, // Set commands must be encrypted
+      commandByte: 7, // 0x07
```

Comment:
Could we introduce an "enum" ala https://github.com/kueblc/mocktuyacloud/blob/master/lib/lan-frame.js#L2 for readability and convenience? Could also use `const`s.

Reply:
Forgive me for not understanding, but why would a structure like that be better? You'd still have to set the same properties.

Reply:
It wouldn't make a functional difference but IMO it would be easier on the programmer to understand the significance of the values `0x07`, `0x0a` etc, and also provide a reference as to the available commands.

The same line could read `commandByte: CommandType.CONTROL,` and then its immediately obvious what the property means. You also have a sense of the other values you could pass to activate other functions of the device, ie, performing device upgrades or relocating the device to a new WiFi network.

Reply:
Ok, that makes more sense now.  Is it ok if I copy the list of control bytes from mocktuyacloud?

Reply:
Absolutely, go for it

Reply:
Added the enum, it does look a lot better now.